libs/hpc-data/src/lib/flows.ts

```diff
@@ -1,228 +1,523 @@
+    firstReportedDate: DATE_FROM_STRING,
+    flowDate: DATE_FROM_STRING,
+    flowObjects: t.array(CREATE_FLOW_OBJECT),
+    isCancellation: optional(t.boolean),
+    newCategories: t.array(t.number),
+    newMoney,
+    origCurrency,
+    parents: t.array(PARENT_TYPE),
+    reportDetails: t.array(CREATE_FLOW_REPORT_DETAIL),
+    restricted,
   }),
   t.partial({
-    budgetYear: t.union([t.string, t.null]),
-    origAmount: t.union([t.string, t.null]),
-    origCurrency: t.union([t.string, t.null]),
-    exchangeRate: t.union([t.string, t.null]),
-    versionEndDate: t.union([t.string, t.null]),
-    deletedAt: t.union([t.string, t.null]),
-    legacy: t.union([
+    notes,
+    pendingStatus: t.union([t.boolean, t.array(t.string)]),
+    cancelled: optional(t.boolean),
+    childMethod: t.union([t.string, CHILD_METHOD_TYPE]),
+    planEntities: t.union([t.boolean, t.array(t.string)]),
+    planIndicated: t.union([t.boolean, t.array(t.string)]),
+    isApprovedFlowVersion: optional(t.boolean),
+    isErrorCorrection: optional(t.boolean),
+    inactiveReason: t.array(FLOW_FORM_FIELD),
+    rejected: optional(t.boolean),
+    versions: t.array(
       t.type({
-        createdAt: t.string,
-        legacyID: t.number,
-        objectID: t.number,
-        objectType: t.string,
-        updatedAt: t.string,
-      }),
-      t.null,
-    ]),
+        id: t.number,
+        versionID: t.number,
+        activeStatus: t.union([t.boolean, t.undefined]),
+        isPending: t.boolean,
+        isCancelled: t.boolean,
+      })
+    ),
+    budgetYear: t.number,
+    origAmount,
+    exchangeRate,
+    versionStartDate,
+    versionEndDate,
   }),
 ]);
 
-export type FlowREST = t.TypeOf<typeof FLOW_REST>;
+export const CREATE_FLOW_PARAMS = t.type({
+  flow: CREATE_FLOW,
+});
 
-export const GET_FLOW_PARAMS = t.type({
-  id: POSITIVE_INTEGER_FROM_STRING,
+export type CreateFlowParams = t.TypeOf<typeof CREATE_FLOW_PARAMS>;
+
+const {
+  lastUpdatedBy: _lastUpdatedBy,
+  createdBy: _createdBy,
+  versions: _versions,
+  legacy: _legacy,
+  ...otherGetFlowProps
+} = GET_FLOW_RESULT.props;
+
+const { categories: _categories, ...createFlowOrganizationProps } =
+  GET_FLOW_ORGANIZATION.props;
+export const CREATE_FLOW_RESULT = t.type({
+  ...otherGetFlowProps,
+  // `organizations` on create flow doesn't contain `categories`
+  organizations: t.array(t.type(createFlowOrganizationProps)),
 });
 
-export type GetFlowParams = t.TypeOf<typeof GET_FLOW_PARAMS>;
+type CreateFlowResult = t.TypeOf<typeof CREATE_FLOW_RESULT>;
 
-export const GET_FLOW_RESULT = FLOW_REST;
+export const UPDATE_FLOW_PARAMS = t.type({
+  flow: t.intersection([
+    CREATE_FLOW,
+    t.type({
+      id: t.number,
+      versionID: t.number,
+    }),
+  ]),
+});
 
-export type GetFlowResult = t.TypeOf<typeof GET_FLOW_RESULT>;
+export type UpdateFlowParams = t.TypeOf<typeof UPDATE_FLOW_PARAMS>;
+
+// * GRAPHQL CODE FROM HERE *
```

Comment:
You may want to try regions like this
```suggestion
// #region GraphQL code
```
Some other languages have this as a native feature (preprocessors) and editors allow you to collapse the code between `#region` and `#endregion` directives. In JavaScript, there is no similar feature in the language itself, but popular editors like VS Code allow you to have collapsible regions by using such comments. I don't like how it gets huge in the scrollbar on the side.

Although, it doesn't seem to be true that all code starting from here is for GraphQL API. I wanted to see how you use these types with GraphQL and I tried `BulkRejectPendingFlowsParams`, which isn't for GraphQL endpoint, but REST.

Also, some types like `NestedFlowFilters` (which were added before) aren't used anywhere. I'd say the only types/codecs truly used for GraphQL code are those appearing when you search for `callGraphQL`, which there are only 3 (should be times two, for params and results), but not as many as you have under this comment.

Reply:
Nice to know about `#region`! I think this comment lost its purpose as you said, so I think I'll just remove it. This was added when I was adding code along side out other senior frontend dev and I didn't want to touch his code